docs/cryptodoc/src/10_tpm.rst

```diff
@@ -0,0 +1,355 @@
+
+   **Input:**
+
+   - ctx: The TPM context
+   - sessions: The TPM session bundle
+   - auth_value: The TPM authorization value for the key
+   - parent_key: The parent key under which the new key shall be created
+   - key_spec: RSA keylength and exponent or ECC curve spec
+
+   **Output:**
+
+   - key: A transient private key object
+
+   **Steps:**
+
+   1. Create a ``TPM2B_SENSITIVE_CREATE`` structure with ``auth_value``
+   2. Create a ``TPMT_PUBLIC`` key template that does not restrict the key for any specific use case
+   3. ``pub_info, priv_bytes = Esys_CreateLoaded(ctx, parent_key, sessions, sensitive_data, template)``
+   4. Return a ``TPM2::PrivateKey`` as a wrapper object
+
+.. admonition:: Transient Key loading
+
+   **Code:**
+
+   - :srcref:`TPM2::PrivateKey::load_transient <src/lib/prov/tpm2/tpm2_key.cpp:190|load_transient>`
+
+   **Input:**
+
+   - ctx: The TPM context
+   - auth_value: The TPM authorization value for the key
+   - parent_key: The parent key under which the new key shall be created
+   - public_blob: The public part of the key
+   - private_blob: The private part of the key
+   - sessions: The TPM session bundle
+
+   **Output:**
+
+   - key: A transient private key object
+
+   **Steps:**
+
+   1. ``handle = Esys_Load(ctx, parent_key, sessions, public_blob, private_blob)``
+   2. ``Esys_TR_SetAuth(ctx, handle, auth_value)``
+   3. Return a ``TPM2::PrivateKey`` as a wrapper object
+
+.. admonition:: Persistent Key loading
+
+   **Code:**
+
+   - :srcref:`TPM2::PrivateKey::load_persistent <src/lib/prov/tpm2/tpm2_key.cpp:180|load_persistent>`
+
+   **Input:**
+
+   - ctx: The TPM context
+   - persistent_handle: The handle of the persistent key to load
+   - auth_value: The TPM authorization value for the key
+   - sessions: The TPM session bundle
+
+   **Output:**
+
+   - key: A persistent private key object
+
+   **Steps:**
+
+   1. ``handle = Esys_TR_FromTPMPublic(ctx, persistent_handle, sessions)``
+   2. ``Esys_TR_SetAuth(ctx, handle, auth_value)``
+   3. Return a ``TPM2::PrivateKey`` as a wrapper object
+
+
+Signature Generation and Verification
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Signatures are supported for both RSA and ECC keys. The implementation is
+largely the same for both wrappers. Therefore, we provide a unified description
+here.
+
+.. admonition:: Signature Generation
+
+   **Input:**
+
+   - ctx: The TPM context
+   - key: The TPM private key
+   - sessions: The TPM session bundle
+   - hash_name: The hash algorithm to use for the signature
+   - data: The data to sign
+
+   **Output:**
+
+   - signature: The signature of the data
+
+   **Steps:**
+
+   1. Calculate the digest of ``data``:
+
+      1. If ``key`` is *not marked* as "restricted", use Botan's software implementation of ``hash_name``
+      2. Otherwise, use the TPM to calculate the digest (see :srcref:`[src/lib/prov/tpm2]/tpm2_hash.cpp`):
+
+         1. ``hash_obj = Esys_HashSequenceStart(ctx, sessions, hash_type)``
```

Comment:
```suggestion
         1. ``hash_obj = Esys_HashSequenceStart(ctx, sessions, hash_name)``
``` 
I know it's technically not correct. But I think this abstraction is fine for the doc

Reply:
For sure. That's also what I meant to write. Thanks.